packages/connect/src/api/bitcoin/refTx.ts

```diff
@@ -78,153 +78,219 @@ const enhanceTransaction = (refTx: RefTransaction, srcTx: BitcoinJsTransaction)
+        script_sig: input.script.toString('hex'),
+    });
+
+    // map bin_outputs, required by TXOUTPUT (TxAckPrevOutput) request from Trezor
+    const binOutputsMap = (output: BitcoinJsOutput) => ({
+        amount: output.value.toString(),
+        script_pubkey: output.script.toString('hex'),
+    });
+
+    const refTx: RefTransaction = {
+        version: tx.version,
+        hash: tx.getId(),
+        inputs: tx.ins.map(inputsMap),
+        bin_outputs: tx.outs.map(binOutputsMap),
+        lock_time: tx.locktime,
+        timestamp: tx.timestamp,
+        expiry: tx.expiry,
+    };
+
+    return enhanceTransaction(refTx, tx);
+};
+
 export const transformReferencedTransactions = (
     txs: TypedRawTransaction[],
     coinInfo: CoinInfo,
 ): RefTransaction[] =>
-    txs.flatMap(raw => {
-        if (coinInfo.type !== 'bitcoin' || raw.type !== 'blockbook') return [];
-        const { hex } = raw.tx;
-        const tx = BitcoinJsTransaction.fromHex(hex, { network: coinInfo.network });
+    coinInfo.type !== 'bitcoin'
+        ? []
+        : txs.filter(isTransactionWithHex).map(raw => {
+              const tx = BitcoinJsTransaction.fromHex(raw.tx.hex, { network: coinInfo.network });
+              return transformReferencedTransaction(tx);
+          });
 
-        // inputs, required by TXINPUT (TxAckPrevInput) request from Trezor
-        const inputsMap = (input: BitcoinJsInput) => ({
-            prev_index: input.index,
-            sequence: input.sequence,
-            prev_hash: bufferUtils.reverseBuffer(input.hash).toString('hex'),
-            script_sig: input.script.toString('hex'),
-        });
+const transformAccountToOrigTransaction = ({
+    tx,
+    addresses,
+    coinInfo,
+}: {
+    tx: AccountTransaction;
+    coinInfo: CoinInfo;
+    addresses?: AccountAddresses;
+}) => {
+    if (!tx.hex)
+        throw TypedError('Method_InvalidParameter', `refTx: hex for ${tx.txid} not provided`);
+    if (!addresses)
+        throw TypedError('Method_InvalidParameter', `refTx: addresses for ${tx.txid} not provided`);
 
-        // map bin_outputs, required by TXOUTPUT (TxAckPrevOutput) request from Trezor
-        const binOutputsMap = (output: BitcoinJsOutput) => ({
-            amount: output.value.toString(),
-            script_pubkey: output.script.toString('hex'),
-        });
+    const srcTx = BitcoinJsTransaction.fromHex(tx.hex, { network: coinInfo.network });
+    return transformOrigTransaction({ ...tx.details, txid: tx.txid }, srcTx, addresses);
+};
 
-        const refTx: RefTransaction = {
-            version: tx.version,
-            hash: tx.getId(),
-            inputs: tx.ins.map(inputsMap),
-            bin_outputs: tx.outs.map(binOutputsMap),
-            lock_time: tx.locktime,
-            timestamp: tx.timestamp,
-            expiry: tx.expiry,
-        };
+const transformAccountToReferencedTransaction = ({
+    tx,
+    coinInfo,
+}: {
+    tx: AccountTransaction;
+    coinInfo: CoinInfo;
+}) => {
```

Comment:
the same micro-nitpick. my personal preference is having return types verbosely listed for data transformation methods. feel free to dismiss

Reply:
[10a24a5](https://github.com/trezor/trezor-suite/pull/9040/commits/10a24a50842e2cad020670556d6bb503ffd5ebeb)